src/seedu/addressbook/data/person/Address.java

```diff
@@ -3,57 +3,89 @@
+ * Represents a Person's address in the address book. Guarantees: immutable; is
+ * valid as declared in {@link #isValidAddress(String)}
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
-
-    public final String value;
-    private boolean isPrivate;
-
-    /**
-     * Validates given address.
-     *
-     * @throws IllegalValueException if given address string is invalid.
-     */
-    public Address(String address, boolean isPrivate) throws IllegalValueException {
-        String trimmedAddress = address.trim();
-        this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
-        }
-        this.value = trimmedAddress;
-    }
-
-    /**
-     * Returns true if a given string is a valid person address.
-     */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
-    }
-
-    @Override
-    public String toString() {
-        return value;
-    }
-
-    @Override
-    public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
-    }
-
-    @Override
-    public int hashCode() {
-        return value.hashCode();
-    }
-
-    public boolean isPrivate() {
-        return isPrivate;
-    }
+	public static final String EXAMPLE = "123, some street";
+	public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses should be in the format [BLOCK], [STREET], [UNIT](OPTIONAL)"
+			+ ", [POSTAL CODE](OPTIONAL)";
+	public static final String ADDRESS_VALIDATION_REGEX = ".+";
+	private static final String PS = ", ";
+
+	public Block block;
+	public Street street;
+	public Unit unit;
+	public PostalCode postalCode;
+
+	private boolean isPrivate;
+
+	/**
+	 * Validates given address.
+	 *
+	 * @throws IllegalValueException
+	 *             if given address string is invalid.
+	 */
+	public Address(String address, boolean isPrivate) throws IllegalValueException {
+		String trimmedAddress = address.trim();
+		this.isPrivate = isPrivate;
+		if (!isValidAddress(trimmedAddress)) {
+			throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+		}
+		initialiseValues(trimmedAddress);
+	}
+
+	private void initialiseValues(String address) {
+		String[] addressParts = address.split(",");
+		block = new Block(addressParts[0].trim());
+		if (addressParts.length > 1){street = new Street(addressParts[1].trim());}
+		else street = new Street("");
+		if (addressParts.length > 2){unit = new Unit(addressParts[2].trim());}
```

Comment:
You can remove the magic numbers here. Moreover, these code obviously violate the coding standard.